docs/develop/go/message-passing.mdx

```diff
@@ -213,6 +213,14 @@ func GreetingWorkflow(ctx workflow.Context) error {
 To send Queries, Signals, or Updates, you call methods on a Temporal [Client](https://pkg.go.dev/go.temporal.io/sdk/client#Client).
 To check the argument types required when sending messages -- and the return type for Queries and Updates -- refer to the corresponding handler method in the Workflow Definition.
 
+:::warning Using Continue-as-New and Updates
+
+- Temporal _does not_ support Continue-as-New functionality within Update handlers.
+- Complete all handlers _before_ using Continue-as-New.
+- Use Continue-as-New from your main Workflow Definition method, just as you would complete or fail a Workflow Execution.
```

Comment:
```suggestion
- Use Continue-as-New from your main Workflow function, just as you would complete or fail a Workflow Execution.
```

In Go, workflows are represeted as functions, not methods on structs

Reply:
Applied in follow-up quick patch.